Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -610,6 +568,137 @@ public string PreviewHotkey
+            }
+        }
+
+        private void ToggleInternalPreview()
+        {
+            if (!PreviewVisible)
+            {
+                ShowInternalPreview();
+            }
+            else
+            {
+                HideInternalPreview();
+            }
+        }
+
+        private void ToggleExternalPreview(string path)
+        {
+            _ = QuickLookHelper.ToggleQuickLookAsync(path).ConfigureAwait(false);
+        }
+
+        private void OpenExternalPreview(string path)
+        {
+            _ = QuickLookHelper.OpenQuickLookAsync(path).ConfigureAwait(false);
+        }
+        
+        private void CloseExternalPreview()
+        {
+            _ = QuickLookHelper.CloseQuickLookAsync().ConfigureAwait(false);
+        }
+
+        private void SwitchExternalPreview(string path)
+        {
+            // Switches preview content
+            // When external is off, do nothing
+            _ = QuickLookHelper.SwitchQuickLookAsync(path).ConfigureAwait(false);
+        }
+
+        private void ShowInternalPreview()
+        {
+            ResultAreaColumn = 1;
+            PreviewVisible = true;
+            Results.SelectedItem?.LoadPreviewImage();
+        }
+
+        private void HideInternalPreview()
+        {
+            ResultAreaColumn = 3;
+            PreviewVisible = false;
+        }
+
+        public void ResetPreview()
+        {
+            if (Settings.AlwaysPreview == true && !PreviewVisible)
+            {
+                ShowInternalPreview();
+            }
+            else
+            {
+                HideInternalPreview();
+            }
+        }
+        
+        private void UpdatePreview()
+        {
+            if (Settings.UseExternalPreview)
+            {
+                if (CanExternalPreviewSelectedResult(out var path))
+                {
+                    // Should use external preview for selected result
+                    if (PreviewVisible)
+                    {
+                        // Previewing
+                        // When internal is open and select a result that should use external preview
+                        // External must be off when PreviewVisible
+                        HideInternalPreview();
+                        OpenExternalPreview(path);
+                    }
+                    else
+                    {
+                        // Internal is off, try to switch preview content
+                        SwitchExternalPreview(path);
+                    }
+                }
+                else
+                {
+                    // Should use internal preview for selected result
+                    if (PreviewVisible)
+                    {
+                        Results.SelectedItem?.LoadPreviewImage();
+                    }
+                    else
+                    {
+                        CloseExternalPreview(); // Forcibly close, ideally should only close when it's on
```

Comment:
We should detect QuickLook status here. If it's on then close it and show builtin preview.